Chapter3/listSort.erl

```diff
@@ -0,0 +1,83 @@
+-module(listSort).
+-export([quicksort/1]).
+-export([mergesort/1]).
+-export([mergesort_split_bottom/3]).
+-export([mergesort_split_top/2]).
+% -export([sort/1]).
+% -export([merge/2]).
+
+
+
+quicksort([]) -> [];
+quicksort([Pivot|Tail]) -> quicksort(quick_less(Tail, Pivot)) ++ [Pivot] ++ quicksort(quick_more(Tail, Pivot)).
+
+%%Example of a one line solution I found online
+%quicksort([Pivot|Tail]) -> quicksort([X || X <- Tail, X < Pivot]) ++ [Pivot] ++ quicksort([X || X <- Tail, X >= Pivot]).
+
+
+
+quick_less(List, Max) -> quick_less_output(List, Max, []).
+
+quick_less_output([], _, Output) -> Output;
+quick_less_output([Head|Tail], Max, Output) ->
+	case Head < Max of
+		true -> quick_less_output(Tail, Max, Output ++ [Head]);
+		false -> quick_less_output(Tail, Max, Output)
+	end.
+
+
+quick_more(List, Min) -> quick_more_output(List, Min, []).
+
+quick_more_output([], _, Output) -> Output;
+quick_more_output([Head|Tail], Min, Output) ->
+	case Head > Min of
+		true -> quick_more_output(Tail, Min, Output ++ [Head]);
+		false -> quick_more_output(Tail, Min, Output)
+	end.
+
+
+
+
+
+
+
+
+
+mergesort(List) when length(List) =< 1 -> List;
```

Comment:
There's a shorter version of this expression, which shows this is a list and it contains only 1 single element in it.
```erlang
mergesort([Item]) ->
    List;
```